src/main/java/at/hannibal2/skyhanni/features/combat/end/DragonFeatures.kt

```diff
@@ -0,0 +1,377 @@
+    /** REGEX-TEST: §f             §r§e§l1st Damager §r§7- §r§a[VIP] Jarre07§r§f §r§7- §r§e9,659,033
+     * REGEX-TEST: §f          §r§6§l2nd Damager §r§7- §r§b[MVP§r§9+§r§b] FlamingZoom§r§f §r§7- §r§e1,459,691
+     * REGEX-TEST: §f          §r§c§l3rd Damager §r§7- §r§b[MVP§r§f+§r§b] Dustbringer§r§f §r§7- §r§e1,091,163
+     */
+    private val endLeaderboard by chatGroup.pattern(
+        "end.place",
+        "§f +§r§.§l(?<Position>\\d+).. Damager §r§7- §r§.(?:\\[[^ ]+\\] )?(?<Name>.*)§r§. §r§7- §r§e(?<Damage>[\\d.,]+)",
+    )
+
+    /** REGEX-TEST: §f                       §r§eZealots Contributed: §r§a27§r§e/100
+     */
+    private val endZealots by protectorRepoGroup.pattern(
+        "chat.end.zealot",
+        "§f +§r§eZealots Contributed: §r§a(?<Amount>\\d+)§r§e/100",
+    )
+
+    /** REGEX-TEST: §5☬ §r§d§lThe §r§5§c§lProtector Dragon§r§d§l has spawned!
+     */
+    private val dragonSpawn by chatGroup.pattern(
+        "spawn",
+        "§5☬ §r§d§lThe §r§5§c§l(?<Dragon>$dragonNamesAsRegex) Dragon§r§d§l has spawned!",
+    )
+
+    /** REGEX-TEST: Your Damage: §c2,003.2
+     */
+    private val scoreDamage by scoreBoardGroup.pattern("damage", "Your Damage: §c(?<Damage>[\\w,.]+)")
+
+    /** REGEX-TEST: Dragon HP: §a14,659,354 §c❤
+     */
+    private val scoreDragon by scoreBoardGroup.pattern("dragon", "Dragon HP: .*")
+
+    // private val scoreProtector by protectorRepoGroup.pattern("scoreboard.protector", "Protector HP: .*")
+
+    /** REGEX-TEST:  §r§bJamBeastie: §r§c7.4M❤
+     * REGEX-TEST:  §r§a42069HzMonitor: §r§c3M❤
+     * REGEX-TEST:  §r§bItsJxxxxx2001: §r§c457k❤
+     * REGEX-TEST:  §r§bThunderblade73: §r§c12.3k❤
+     */
+    private val tabDamage by tabListGroup.pattern(
+        "fight.player",
+        ".*§r§.(?<Name>.+): §r§c(?<Damage>[\\d.]+)(?<Unit>[kM])?❤",
+    )
+
+    private var yourEyes = 0
+
+    private var dragonSpawned = false
+        set(value) {
+            field = value
+            if (dragonSpawned) {
+                egg = false
+            }
+        }
+
+    private enum class Type {
+        GOLEM,
+        DRAGON
+    }
+
+    private var endType: Type? = null
+    private var endTopDamage = 0.0
+    private var endDamage = 0.0
+    private var endPlace = 0
+
+    private var currentDamage = 0.0
+    private var currentTopDamage = 0.0
+    private var currentPlace: Int? = null
+    private var widgetActive = false
+    var egg = true
+    var weight = 0.0
+    private var currentDragonType: DragonType? = null
+
+    private fun resetEnd() {
+        endType = null
+        endTopDamage = 0.0
+        endDamage = 0.0
+        endPlace = 0
+    }
+
+    private fun reset() {
+        resetEnd()
+        dragonSpawned = false
+        currentTopDamage = 0.0
+        currentDamage = 0.0
+        currentPlace = null
+        widgetActive = false
+        yourEyes = 0
+        currentDragonType = null
+    }
+
+    private fun enable() = LorenzUtils.inSkyBlock && IslandType.THE_END.isInIsland()
```

Comment:
```suggestion
    private fun isEnabled() = LorenzUtils.inSkyBlock && IslandType.THE_END.isInIsland()
```

_However_, you can get rid of the necessity for this and the `enableDisplay` function entirely by using `@HandleEvent` parameters.

```diff
-    @HandleEvent
-    fun onChat(event: SkyHanniChatEvent) {
-        if (!enable()) return
+   @HandleEvent(onlyOnSkyblock = true, onlyOnIsland = IslandType.THE_END)
+   fun onChat(event: SkyHanniChatEvent) {
+       if (!config.display) return
```